wordpress-mobile/WordPress-iOS
# Domain credit success UI

Refs. #11467 - subtask - add success UI

* Set register domain row `showsSelectionState` to false, because its action presents a modal.
* Added `attributedSubtitleConfiguration` parameter in `NoResultsViewController` constructor/configuration to allow caller to further configure subtitle attributed text after default styling.
  * Passed `nil` value in Objective-C callers.
  * This addition is to allow different text style of domain string on success screen.
* Created `DomainCreditRedemptionSuccessViewController` that displays congrats messaging after domain credit redemption, with delegate to handle continue button action. It reuses UI of `NoResultsViewController` by using it as a child view controller.
* On site dashboard, on tapping "Register Domain" row, showed the new `DomainCreditRedemptionSuccessViewController`. After continue button is pressed on success screen, the screen is dismissed and a snackbar message with more details about domain registration is shown on site dashboard.

To test:
* Turn on `domainCredit` feature flag in `FeatureFlag` enum. You can also change domain name on success screen in `showDomainCreditRedemption`
* Go to a site/blog dashboard
* Tap on "Register Domain" row

Example expected screenshots

device | success screen | after dismissing success screen
--- | --- | ---
iPhone SE | ![Simulator Screen Shot - iPhone SE - 2019-04-26 at 21 19 43](https://user-images.githubusercontent.com/1945542/56811645-c2c52480-686b-11e9-87e8-6c0a3dbbdcf0.png) | ![Simulator Screen Shot - iPhone SE - 2019-04-26 at 21 20 18](https://user-images.githubusercontent.com/1945542/56811646-c35dbb00-686b-11e9-8bc8-6b519b985a23.png)
iPad Air 2 | ![Simulator Screen Shot - iPad Air 2 - 2019-04-26 at 21 45 14](https://user-images.githubusercontent.com/1945542/56812534-ade99080-686d-11e9-9400-2e253475cfb4.png) | ![Simulator Screen Shot - iPad Air 2 - 2019-04-26 at 21 45 18](https://user-images.githubusercontent.com/1945542/56812535-ae822700-686d-11e9-9603-b3a7f68880b5.png)




Update release notes:
